src/main/java/org/opensearch/neuralsearch/processor/chunker/DelimiterChunker.java

```diff
@@ -88,4 +137,24 @@ public List<String> chunk(final String content, final Map<String, Object> runtim
     public String getAlgorithmName() {
         return ALGORITHM_NAME;
     }
+
+    private List<String> splitByLength(final String text, final int maxLength) {
```

Comment:
Could you explain a little about this method?
if text is `abcde<tag>`, delimiter is `<tag>`, and maxLength is 3, this will return `abcde<tag>` which exceed maxLength.